linux/Jenkinsfile

```diff
@@ -131,69 +139,92 @@ pipeline {
+    env.COMPOSE_DOCKER_CLI_BUILD=1
+    env._JAVA_OPTIONS =  (buildArch == "armv7l" && DISTRO == "Debian")? "" : "-Xmx4g"
 }
 
-def buildAndTest(DISTRO, arch) {
-    if (DISTRO != "Debian") {
-        // Install Adoptium GPG key for RPM signing
-        withCredentials([file(credentialsId: 'adoptium-artifactory-gpg-key', variable: 'GPG_KEY')]) {
-            sh('./gradlew packageJdk${DISTRO} checkJdk${DISTRO} --parallel -PPRODUCT=${PRODUCT} -PPRODUCT_VERSION=${VERSION} -PGPG_KEY=${GPG_KEY}')
+def buildAndTest(DISTRO, buildArch) {
+    try {
+        if (DISTRO != "Debian") { // for RPM based: RedHat / Suse
+            // Install Adoptium GPG key for RPM signing
+            withCredentials([file(credentialsId: 'adoptium-artifactory-gpg-key', variable: 'GPG_KEY')]) {
+                def buildCLI = "./gradlew packageJdk${DISTRO} checkJdk${DISTRO} --parallel -PPRODUCT=${env.PRODUCT} -PPRODUCT_VERSION=${VERSION} -PGPG_KEY=${GPG_KEY} -PARCH=${buildArch}"
+                buildCLI = params.enableDebug.toBoolean() ? buildCLI + " --stacktrace" : buildCLI
+                sh("$buildCLI")
+            }
+        } else {
+            def gBuildTask = (buildArch == 'x86_64') ? "packageJdk${DISTRO} checkJdk${DISTRO}" : "packageJdk${DISTRO}"
+            def debArchList = [
+                "x86_64" : "amd64",
+                "armv7l": "armhf",
+                "aarch64": "arm64",
+                "ppc64le": "ppc64el",
+                "s390x"  : "s390x"
+            ]
+            def buildCLI = "./gradlew ${gBuildTask} --parallel -PPRODUCT=${env.PRODUCT} -PPRODUCT_VERSION=${VERSION} -PARCH=${debArchList[buildArch]}"
+            buildCLI = params.enableDebug.toBoolean() ? buildCLI + " --stacktrace" : buildCLI
+            sh("$buildCLI")
         }
-    } else if (arch == 'x64') {
-        sh("./gradlew packageJdk${DISTRO} checkJdk${DISTRO} --parallel -PPRODUCT=${PRODUCT} -PPRODUCT_VERSION=${VERSION}")
-    } else {
-        // Current tests are not supported on non-x64 platform
-        sh("./gradlew packageJdk${DISTRO} --parallel -PPRODUCT=${PRODUCT} -PPRODUCT_VERSION=${VERSION}")
+    } catch (Exception ex) {
+        echo 'Exception in buildAndTest: ' + ex.toString() 
+        currentBuild.result = 'FAILURE'  // set the whole pipeline 'red' if build or test fail. Do not use "error" that will not call "finally"
+    } finally {
+        archiveArtifacts artifacts: '**/build/ospackage/*,**/build/reports/**,**/packageTest/dependencies/deb/*', onlyIfSuccessful:false, allowEmptyArchive: true
     }
-    archiveArtifacts artifacts: '**/build/ospackage/*,**/build/reports/**,**/packageTest/dependencies/deb/*'
 }
 
-def uploadArtifacts(distro, arch) {
-    if (distro == "Debian") {
-        uploadDebArtifacts(arch)
-    } else {
-        uploadRPMArtifacts(distro)
+def uploadArtifacts(DISTRO, buildArch) {
+    if (DISTRO == "Debian") {
+        uploadDebArtifacts(buildArch)
+    } else {  // DISTRO == RPM(RedHat||Suse)
```

Comment:
could be slightly brittle, maybe have an else if for this and then and else at the end which errors if distro is not valid